File: src/pcl-bindings/PCDWriter.cpp
#include "PCDWriter.h"

PCDWriter* io_pcdwriter_ctor()
{
	return new PCDWriter();
}

void io_pcdwriter_delete(PCDWriter** ptr)
{
	delete* ptr;
	*ptr = NULL;
}

int32_t io_pcdwriter_write_xyz(PCDWriter* ptr, const char* str, PointCloud<PointXYZ>* cloud, int binary)
{
	return ptr->write(string(str), *cloud, binary);
}

File: src/Pcl.NET/PCDWriter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Pcl.NET
{
  internal class PCDWriter : UnmanagedObject
  {
    public PCDWriter()
    {
      _ptr = Invoke.io_pcdwriter_ctor();
    }

    public void Write(string fileName, PointCloud<PointXYZ> cloud, bool binary = true)
    {
      int err = Invoke.io_pcdwriter_write_xyz(_ptr, fileName, cloud, binary ? 1 : 0);

#warning TODO check err
    }

    protected override void DisposeObject()
    {
      Invoke.io_pcdreader_delete(ref _ptr);
    }
  }
}
